GMnetENGINE.gmx/scripts/htme_clientStart.gml

```diff
@@ -37,6 +37,31 @@ if( self.socketOrServer<0 ){
     return false;
 }
 
+// Get server port from list
+if server_port=0 and udphp_provide_server_port
+{
+    var l = global.udphp_downloadlist;
+    if (ds_exists(l,ds_type_list)) 
+    {
+        for (var i=0; i<ds_list_size(l); i+=1)
+        {
+            var entry = l[| 0];
+            var ip = entry[? "ip"];
+            var port_provided = entry[? "data8"];

```

Comment:
Yes that's why I used a option to include it. But if you think of rebuilding a new gate punch in node.js.

*add to personal to do list:
1# Learn what node.js is
2# Learn some node.js